crates/hotshot/src/traits/networking/web_server_network.rs

```diff
@@ -799,15 +799,13 @@ impl<TYPES: NodeType + 'static> ConnectedNetwork<Message<TYPES>, TYPES::Signatur
                     debug!("Somehow task already existed!");
                 }
 
-                // GC proposal collection if we are two views in the future
-                if let Some((_, sender)) = task_map.remove_entry(&view_number.wrapping_sub(2)) {
-                    // Send task cancel message to old task
-
-                    // If task already exited we expect an error
-                    let _res = sender
-                        .send(ConsensusIntentEvent::CancelPollForProposal(
-                            view_number.wrapping_sub(2),
-                        ))
+                // Remove all entries in the task map that are polling for a view less than or equal to `view_number - 2`.
+                let view_minus_2 = view_number.saturating_sub(2);
+                let range = task_map.range(..view_minus_2);
+                for (view, task) in range {
+                    // Cancel the old task by sending a message to it. If the task already exited we expect an error
+                    let _res = task
+                        .send(ConsensusIntentEvent::CancelPollForProposal(*view))
```

Comment:
nit Can we make this a function that takes the event, maybe as a fn like `event_fn: fn(view: u64) -> ConsensusIntentEvent` and then the function could just do range.map(event_fn).  Mostly just making this a function so this code isn't copy pasted a bunch of times